src/shared/metadata.ts

```diff
@@ -0,0 +1,175 @@
+import { APPLICATION } from './config'
+import axios from 'axios'
+
+/**
+ * https://hasura.io/docs/latest/graphql/core/api-reference/schema-metadata-api/relationship.html
+ * Here we are using the schema-metadata-api to track the relationships between auth tables
+ **/
+
+interface Table {
+  name: string
+  schema: string
+  foreignKey?: string
+}
+
+interface Relationship {
+  name: string
+  source: Table
+  destination?: Table
+  isArray?: boolean
+}
+
+function trackTable(table: Table) {
+  return axios.post(
+    APPLICATION.HASURA_ENDPOINT.replace('/v1/graphql', '/v1/query'),
```

Comment:
Is this endpoint available for Hasura v2 too?

Reply:
Not sure why this was resolved by @komninoschat. 

> Supported from
> The metadata API is supported for versions v2.0.0 and above and replaces the older schema/metadata API.

@elitan If by v2 you mean the Metadata API (https://hasura.io/docs/latest/graphql/core/api-reference/metadata-api/table-view.html#pg-track-table), then it is only available starting on v2.0.0. For backwards compatibility, it is better to start using v1 and change it to use the Metadata API instead for the next major release of HBP.